Library/Homebrew/formulary.rb

```diff
@@ -123,7 +123,16 @@ def initialize(bottle_name)
     end
 
     def get_formula(spec, alias_path: nil)
-      formula = super
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
+      path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
+      contents = Utils.popen_read "tar", "-xO", "-f", @bottle_filename,
+        "--strip-components", "3", path, err: :close
+      formula = if $CHILD_STATUS.success?
+        Formulary.from_contents name, HOMEBREW_CELLAR/path, contents, spec
+      else
+        super
```

Comment:
If this stays as-is: worth printing a `Warning` message in this case?

Reply:
I've added a warning.